src/xrdok.js

```diff
@@ -681,3 +681,280 @@ const XRiconchevron = 'xr-icon-chevron';
+      legendEl.appendChild(createLabel(-sec, prim, j, 0.8, data.descriptors[j], data.color));
+    }
+
+    // create grid
+    for (var n = Math.floor(minVal); n <= Math.ceil(maxVal); n++) {
+      const line = document.createElement('a-entity');
+      line.setAttribute('line',
+        `start: ${sec} ${n-offset} 0; 
+         end: ${sec} ${n-offset} -${count};
+         color: white;
+        `
+      );
+      gridEl.appendChild(line);
+    }
+
+    legendEl.appendChild(gridEl);
+    legendEl.setAttribute('visible', data.showLegend);
+
+    return legendEl;
+  }
+
+  function createGeometryLine(data) {
+    var offset = data.primaryValues[0];
+    var lineGraphEl = document.createElement('a-entity');
+    lineGraphEl.setAttribute('id', 'line-graph-parent');
+    var lineGraphGeo = new THREE.Geometry();
+    var lineGraphMat = new THREE.LineBasicMaterial({
+      color: data.color,
+      linewidth: 15,
+    });
+    const sec = data.secondaryValuesBase;
+
+    // create geometry
+    for (var j=0; j < Math.min(data.descriptors.length, data.primaryValues.length); j++) {
+      //get const 
+      const prim = data.primaryValues[j]-offset;
+
+      // create vertices
+      lineGraphGeo.vertices.push(
+        new THREE.Vector3(sec, prim, -j)
+      );
+    }
+    
+    // create mesh
+    var lineGraph = new THREE.Line(lineGraphGeo, lineGraphMat);
+    lineGraphEl.setObject3D('xr-line-graph', lineGraph);
+    lineGraphEl.setAttribute('visible', data.showLine);
+
+    return lineGraphEl;
+  }
+
+  function createGeometryPlane(data) {
+
+    var offset = data.primaryValues[0];
+    var planeGraphEl = document.createElement('a-entity');
+    planeGraphEl.setAttribute('id', 'plane-graph-parent');
+    var planeGraphPrimaryEl = document.createElement('a-entity');
+    planeGraphPrimaryEl.setAttribute('id', 'plane-graph-primary-parent');
+    var planeGraphSecondaryEl = document.createElement('a-entity');
+    planeGraphSecondaryEl.setAttribute('id', 'plane-graph-secondary-parent');
+    var planeGraphPrimaryGeo = new THREE.Geometry();
+    var planeGraphPrimaryMat = new THREE.MeshStandardMaterial();
+    var planeGraphSecondaryGeo = new THREE.Geometry();
+    var planeGraphSecondaryMat = new THREE.MeshStandardMaterial();
+    
+    // create geometry
+    for (var j = 0; j < Math.min(data.descriptors.length, data.primaryValues.length); j++) {
+      // get const
+      const prim = data.primaryValues[j]-offset;
+      const sec = (data.secondaryValuesBase * data.secondaryValues[j]/data.secondaryValuesBase)/100;
+      const linePath = document.createElement('a-entity');
+
+      // create primary vertices
+      planeGraphPrimaryGeo.vertices.push(
+        new THREE.Vector3(-data.secondaryValuesBase, prim, -j),
+        new THREE.Vector3(data.secondaryValuesBase, prim, -j)
+      );
+
+      // create secondary vertices
+      planeGraphSecondaryGeo.vertices.push(
+        new THREE.Vector3(-sec, prim + 0.001, -j),
```

Comment:
Might even need a 0.002 offset, since there is a chance of flickering still to be seen.